File: consumerNodejs/src/model/message.ts
import mongoose from "mongoose";
import {createId}  from "@paralleldrive/cuid2"

const messageSchema = new mongoose.Schema(
  {
    identifier: {type: String, required: true, unique: true},
    phone_number: { type: String, required: true },
    message: { type: String, required: true },
    campaign_id: { type: String, required: true },
    deleted: {type: Boolean, required: true}

  },
  { timestamps: true }
);

messageSchema.set("timestamps", {
  createdAt: "created_at",
  updatedAt: "updated_at"
})

export const MessageModel = mongoose.model('Message', messageSchema);

messageSchema.pre("save", (next,opts)=>{
    const message = this as any
    try{
        message.identifier = createId()
        next()
    }catch(e){
        console.error("ERR_TO_SAVE_MESSAGE",e)
        throw new Error(e)
    }
})

File: consumerGo/internal/mongo/models/message.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Message struct {
	ID          primitive.ObjectID `bson:"_id,omitempty"`
	Identifier  string             `bson:"identifier,omitempty"`
	PhoneNumber string             `bson:"phone_number,omitempty"`
	Message     string             `bson:"message,omitempty"`
	CampaignID  string             `bson:"campaign_id,omitempty"`
	Deleted     bool               `bson:"deleted"`
	CreatedAt   primitive.DateTime `bson:"created_at,omitempty"`
	UpdatedAt   primitive.DateTime `bson:"updated_at,omitempty"`
}
